lib/Transforms/IPO/StructFieldAccessInfo.cpp

```diff
@@ -0,0 +1,349 @@
+// lib/Tranforms/IPO/StructFieldAccessInfo.cpp - Implements class of
+// StructFieldAccessInfo -*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===------------------------------------------------------------------------===//
+//
+// This file implements StructFieldAccessInfo that is used in
+// StructFieldCacheAnalysis pass.
+//
+//===------------------------------------------------------------------------===//
+
+#include "StructFieldCacheAnalysisImpl.h"
```

Comment:
I don't want to put my header file in include/ so I put the header along with the source code (in lib/) in order to make my classes private. I saw a few other use cases in lib/. Do you think it's a bad idea?